src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -56,15 +56,11 @@ var validUTCDateTime = function (dateTimeComposition){
 var validString = function (dateTimeString) {
     if (!dateTimeString || dateTimeString.toString() === '')
         return false;
-    return true;
-};
-
-// check the characters of string are digits
-var validDigitString = function (dateTimeString) {
+    // datetime format in the spec: YYYY[MM[DD[HH[MM[SS[.S[S[S[S]]]]]]]]][+/-ZZZZ],
     var ds = dateTimeString.toString();
-    if (!/^(\d+)$/.test(ds))
-        throw `Contains non-numeric characters in ${ds}`;
-    return ds;
+    if (!/^((\d{4}(\d{2}(\d{2}(\d{2}(\d{2}(\d{2}(\.\d+)?)?)?)?)?)?((-|\+)\d{1,4})?))$/.test(ds))
```

Comment:
Nit: You have an extra set of parenthesis for the group between the ^ and $.

Reply:
Thanks~